File: production/frontend/src/api/refinance.ts
/**
 * api/refinance.ts — MOD-11 再融资/再贴现闭环 API
 *
 * 后端契约: backend/app/api/v1/modules/refinance.py
 *   GET  /modules/refinance/{enterpriseId}/cashflow     6 个月现金流预测
 *   GET  /modules/refinance/{enterpriseId}/entrance     再融资入口判定 + 缺口规模标签
 *   GET  /modules/refinance/{enterpriseId}/recommend    AI 推荐 4 款再融资产品
 *   POST /modules/refinance/{enterpriseId}/submit?recId=xxx  提交再融资申请
 */

import type { Id, IsoTimestamp, AmountInCents } from '@contracts/common';
import { get, post } from './client';

export type GapSizeLabel = 'small' | 'medium' | 'large';

export type SubmissionStatus =
  | 'draft'
  | 'submitted'
  | 'pending_approval'
  | 'approved'
  | 'rejected';

export interface CashflowForecast {
  enterpriseId: Id;
  monthIso: string;
  projectedInflowCents: AmountInCents;
  projectedOutflowCents: AmountInCents;
  gapCents: number;
  cumulativeGapCents: number;
}

export interface RefinanceEntrance {
  enterpriseId: Id;
  eligible: boolean;
  gapSizeLabel: GapSizeLabel;
  suggestedSchemes: string[];
}

export interface AIRecommendation {
  recId: Id;
  productName: string;
  lender: string;
  amountCents: AmountInCents;
  annualRatePct: number;
  termMonths: number;
  expectedApprovalProb: number;
  totalCostCents: AmountInCents;
  reasons: string[];
}

export interface RefinanceSubmission {
  enterpriseId: Id;
  recId: Id;
  status: SubmissionStatus;
  createdAt: IsoTimestamp;
}

/** 6 个月现金流预测 */
export async function forecastCashflow(
  enterpriseId: Id,
  months = 6,
): Promise<CashflowForecast[]> {
  return get<CashflowForecast[]>('/modules/refinance/' + enterpriseId + '/cashflow', {
    params: { months },
  });
}

/** 再融资入口判定与缺口规模标签 */
export async function computeEntrance(enterpriseId: Id): Promise<RefinanceEntrance> {
  return get<RefinanceEntrance>('/modules/refinance/' + enterpriseId + '/entrance');
}

/** AI 推荐 4 款再融资产品 */
export async function aiRecommend(
  enterpriseId: Id,
  useLlm = true,
): Promise<AIRecommendation[]> {
  return get<AIRecommendation[]>('/modules/refinance/' + enterpriseId + '/recommend', {
    params: { useLlm },
  });
}

/** 提交再融资申请 (指定 recId) */
export async function submitApplication(
  enterpriseId: Id,
  recId: Id,
): Promise<RefinanceSubmission> {
  return post<RefinanceSubmission>(
    '/modules/refinance/' + enterpriseId + '/submit',
    null,
    { params: { recId } },
  );
}

File: production/backend/app/schemas/refinance.py
"""文件名：refinance.py 职责：再融资 Pydantic 模型,定义缺口标签、提交状态、现金流预测与 AI 推荐 schema."""
from __future__ import annotations

from enum import Enum
from typing import Optional

from pydantic import BaseModel, ConfigDict, Field
from pydantic.alias_generators import to_camel

from app.schemas.common import AmountInCents, Id, IsoTimestamp, Ratio


class GapSizeLabel(str, Enum):
    SMALL = "small"
    MEDIUM = "medium"
    LARGE = "large"


class SubmissionStatus(str, Enum):
    DRAFT = "draft"
    SUBMITTED = "submitted"
    PENDING_APPROVAL = "pending_approval"
    APPROVED = "approved"
    REJECTED = "rejected"


class _RefiBase(BaseModel):
    model_config = ConfigDict(
        populate_by_name=True,
        alias_generator=to_camel,
        use_enum_values=True,
    )


class CashflowForecast(_RefiBase):
    enterprise_id: Id
    month_iso: str
    projected_inflow_cents: AmountInCents
    projected_outflow_cents: AmountInCents
    gap_cents: int
    cumulative_gap_cents: int


class RefinanceEntrance(_RefiBase):
    enterprise_id: Id
    eligible: bool
    gap_size_label: GapSizeLabel
    suggested_schemes: list[str] = Field(default_factory=list)


class AIRecommendation(_RefiBase):
    rec_id: Id
    product_name: str
    lender: str
    amount_cents: AmountInCents
    annual_rate_pct: float = Field(ge=0.0)
    term_months: int = Field(ge=1)
    expected_approval_prob: float = Field(ge=0.0, le=1.0)
    total_cost_cents: AmountInCents
    reasons: list[str] = Field(default_factory=list)


class RefinanceSubmission(_RefiBase):
    enterprise_id: Id
    rec_id: Id
    status: SubmissionStatus
    created_at: IsoTimestamp
